### PR TITLE
New Data Source: `azurerm_builtin_role_definition`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Adding a new data source for the built-in Azure role definitions to be able to specify friendly names rather than GUID's

```
$ acctests azurerm TestAccDataSourceAzureRMBuiltInRoleDefinition_
=== RUN   TestAccDataSourceAzureRMBuiltInRoleDefinition_contributor
--- PASS: TestAccDataSourceAzureRMBuiltInRoleDefinition_contributor (8.28s)
=== RUN   TestAccDataSourceAzureRMBuiltInRoleDefinition_owner
--- PASS: TestAccDataSourceAzureRMBuiltInRoleDefinition_owner (7.63s)
=== RUN   TestAccDataSourceAzureRMBuiltInRoleDefinition_reader
--- PASS: TestAccDataSourceAzureRMBuiltInRoleDefinition_reader (7.06s)
=== RUN   TestAccDataSourceAzureRMBuiltInRoleDefinition_virtualMachineContributor
--- PASS: TestAccDataSourceAzureRMBuiltInRoleDefinition_virtualMachineContributor (7.68s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm    30.676s
```

### CODE REVIEW

File: azurerm/data_source_arm_builtin_role_definition.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	name := d.Get("name").(string)
+	roleDefinitionIds := map[string]string{
+		"Contributor":               "b24988ac-6180-42a0-ab88-20f7382dd24c",
+		"Owner":                     "8e3af657-a8ff-443c-a75c-2fe8c4bcb635",
+		"Reader":                    "acdd72a7-3385-48ef-bd42-f606fba81ae7",
+		"VirtualMachineContributor": "d73bb868-a0df-4d4d-bd69-98a00b01fccb",
+	}
+	roleDefinitionId := roleDefinitionIds[name]
+
+	// TODO: when the API's fixed - pull out additional information from the API
```
Comment (tombuildsstuff) [inline azurerm/data_source_arm_builtin_role_definition.go]: I'll open a bug shortly and link that here

File: azurerm/data_source_arm_builtin_role_definition.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	}
+}
+
+func dataSourceArmBuiltInRoleDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	name := d.Get("name").(string)
+	roleDefinitionIds := map[string]string{
+		"Contributor":               "b24988ac-6180-42a0-ab88-20f7382dd24c",
+		"Owner":                     "8e3af657-a8ff-443c-a75c-2fe8c4bcb635",
+		"Reader":                    "acdd72a7-3385-48ef-bd42-f606fba81ae7",
+		"VirtualMachineContributor": "d73bb868-a0df-4d4d-bd69-98a00b01fccb",
```
Comment (tombuildsstuff) [inline azurerm/data_source_arm_builtin_role_definition.go]: these are Constants within Azure - I've confirmed this within the Azure Portal and in a third-party ARM Template.

Comment (radeksimko) [reply azurerm/data_source_arm_builtin_role_definition.go]: Is this documented anywhere publicly - i.e. can we possibly link to any doc page from here?

Comment (tombuildsstuff) [reply azurerm/data_source_arm_builtin_role_definition.go]: Not that I've seen, I found these in an [ARM Template](https://github.com/Azure/azure-quickstart-templates/blob/master/101-rbac-builtinrole-virtualmachine/azuredeploy.json#L40) and confirmed the values in the HTML source of the Azure Portal - so I can confirm they're right.

A [search for the GUID](https://www.google.co.uk/search?8e3af657-a8ff-443c-a75c-2fe8c4bcb635) doesn't bring up anything official from MS - but does confirm it's a global constant ¯\_(ツ)_/¯

**Review by radeksimko — APPROVED** [review]

**Pull request merged**